client/txn.go

```diff
@@ -77,30 +85,200 @@ type txn struct {
 	haveEndTxn   bool // True if there was an explicit EndTransaction
 }
 
-func (t *txn) init(kv *kv) {
-	t.kv = *kv
-	t.wrapped = kv.Sender
-	t.txnSender.txn = t
-	t.kv.Sender = &t.txnSender
+func newTx(db *DB, depth int) *Tx {
+	tx := &Tx{
+		db:      *db,
+		wrapped: db.Sender,
+	}
+	tx.txnSender.Tx = tx
+	tx.db.Sender = &tx.txnSender
```

Comment:
`Tx.txnSender` is the transaction specific sender. A `Tx` internally uses a `DB` in order to send calls using `DB.send`, but it wants that `send` to use the transaction sender. See `DB.send`.